components/scream/src/share/field/field.hpp

```diff
@@ -148,6 +148,10 @@ class Field {
   template<HostOrDevice HD = Device>
   void set_value (const RT value);
 
+  // Copy the data from one field to this field
+  template<HostOrDevice HD = Device>
+  void set_value (const field_type& field_src);
```

Comment:
I would change the name of this method. `set_value` makes me think that you set a single value. I would use something like `assign`, or `import`, or 'deep_copy'. Maybe deep_copy is more inline with what we are doing, and it's the same name kokkos uses.

Reply:
If we go with `deep_copy`, we should also change the name of the method above. Kokkos overloads the `deep_copy` function, offering a copy from view and a copy from scalar value. We would be more aligned with that nomenclature.

Not necessarily in this PR though.